client/src/hooks/useAuth.js

```diff
@@ -1,17 +1,42 @@
 import React, { useState, useEffect, useContext, createContext } from "react";
+import Loading from "components/Loading";
+
 const authContext = createContext();
 
 export function ProvideAuth({ children }) {
   const auth = useProvideAuth();
-  return <authContext.Provider value={auth}>{children}</authContext.Provider>;
+  return auth.isLoading ? (
+    <Loading />
+  ) : (
+    <authContext.Provider value={auth}>{children}</authContext.Provider>
+  );
 }
 
 export const useAuth = () => {
   return useContext(authContext);
 };
 
 function useProvideAuth() {
-  const [user, setUser] = useState({});
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const reAuth = async () => {
+      try {
+        const response = await fetch("/api/users/reAuth", {
+          method: "get",
+        });
+        if (response.ok) {
+          const data = await response.json();
+          if (data.user) setUser(data.user);
+        }
+        setIsLoading(false);
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    reAuth();
+  }, []);
```

Comment:
do you know if this gets called every time you have useAuth() in a component?

Reply:
It doesn't seem to. Only seems to get called once the provider is mounted.